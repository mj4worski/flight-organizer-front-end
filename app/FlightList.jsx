import React from 'react';
import Flight from './Flight.jsx'

const FlightList = (flights) => {
    const flightsPop = flights;
    return (
        <div>
            {
                ( typeof flightsPop.flights == 'undefined' || flightsPop.flights.length === 0 ) ?
                    <p>Nie znaleziono lotow</p> :
                    flightsPop.flights.map(flight =>
                        <Flight flight={flight}/>
                    )
            }
        </div>
    )
};


export default FlightList;