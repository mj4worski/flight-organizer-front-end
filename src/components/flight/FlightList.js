// @flow
import type {FlightType} from '../../state/data-flow'
import React from 'react';
import Flight from './Flight.js'

type FlightList = {
    flights: FlightType,
    reservationFlight: Function,
}

export default ({flights, reservationFlight}: FlightList) => (
    <div>
        {
            ( typeof flights === 'undefined' || flights.length === 0 ) ?
                <p>Nie znaleziono lotow</p> :
                flights.map(flight =>
                    <Flight key={flight.id} flight={flight} onReservationClick={reservationFlight}/>
                )
        }
    </div>
);