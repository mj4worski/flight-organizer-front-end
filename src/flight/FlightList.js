import React from 'react';
import Flight from './Flight.js'

const FlightList = ({flights}) => {
    return (
        <div>
            {
                ( typeof flights === 'undefined' || flights.length === 0 ) ?
                    <p>Nie znaleziono lotow</p> :
                    flights.map(flight =>
                        <Flight key={flight.id} {...flight}/>
                    )
            }
        </div>
    )
};

export default FlightList;