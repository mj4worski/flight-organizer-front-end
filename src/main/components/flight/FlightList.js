import React from 'react';
import PropTypes from 'prop-types';
import Flight from './Flight';

const FlightList = ({ flights }) => (
  <div>{
        (typeof flights === 'undefined' || flights.length === 0) ?
          <p>Nie znaleziono lotow</p> :
            flights.map(flight =>
                (<Flight
                  key={flight.id}
                  flight={flight}
                />),
            )
    }</div>
);

FlightList.propTypes = {
  flights: PropTypes.array,
};

export default FlightList;
