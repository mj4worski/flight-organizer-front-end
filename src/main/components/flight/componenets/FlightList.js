import React from 'react';
import PropTypes from 'prop-types';
import Flight from './Flight';

const FlightList = ({ flights }) => (
  <div>{
        (typeof flights === 'undefined' || flights.length === 0) ?
          <p>Sorry, your current search result doesn&#39;t exist. Try different direction.</p> :
            flights.map(flight =>
                (<Flight
                  key={flight.id}
                  flight={flight}
                />),
            )
    }</div>
);

FlightList.propTypes = {
  flights: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    departureFrom: PropTypes.string,
    arrivalTo: PropTypes.string,
  })),
};

FlightList.defaultProps = {
  flights: [],
};

export default FlightList;
