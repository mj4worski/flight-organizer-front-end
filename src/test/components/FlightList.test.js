import React from 'react';
import renderer from 'react-test-renderer';
import FlightList from '../../main/components/flight/FlightList';

it('should render FlightList with information about non existing flights', () => {
  const component = renderer.create(
    <FlightList />,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render FlightList which contains Flight component', () => {
  const flights = [
    {
      id: 1,
      arrivalTo: 'Paris',
      departureFrom: 'Barcelona',
    },
    {
      id: 2,
      arrivalTo: 'Cracow',
      departureFrom: 'Moscow',
    },
  ];

  const component = renderer.create(
    <FlightList flights={flights} />,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

