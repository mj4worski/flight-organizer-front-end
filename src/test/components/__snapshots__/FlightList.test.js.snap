// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FlightList which contains Flight component 1`] = `
<div>
  <section>
    <h1>
      Znaleziony lot to:
    </h1>
    <p>
      Skad : 
      Barcelona
    </p>
    <p>
      Dokad : 
      Paris
    </p>
    <button
      onClick={[Function]}
    >
      Rezerwacja
    </button>
  </section>
  <section>
    <h1>
      Znaleziony lot to:
    </h1>
    <p>
      Skad : 
      Moscow
    </p>
    <p>
      Dokad : 
      Cracow
    </p>
    <button
      onClick={[Function]}
    >
      Rezerwacja
    </button>
  </section>
</div>
`;

exports[`should render FlightList with information about non existing flights 1`] = `
<div>
  <p>
    Nie znaleziono lotow
  </p>
</div>
`;
